pkg/splunk/enterprise/monitoringconsole.go

```diff
@@ -0,0 +1,343 @@
+
+package enterprise
+
+import (
+	"bytes"
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"reflect"
+	"sort"
+	"strings"
+
+	enterprisev1 "github.com/splunk/splunk-operator/pkg/apis/enterprise/v1alpha3"
+	splcommon "github.com/splunk/splunk-operator/pkg/splunk/common"
+	splctrl "github.com/splunk/splunk-operator/pkg/splunk/controller"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// ApplyMonitoringConsole reconciles the deployment for N standalone instances of Splunk Enterprise.
+func ApplyMonitoringConsole(client splcommon.ControllerClient, cr splcommon.MetaObject, spec enterprisev1.CommonSplunkSpec, extraEnv []corev1.EnvVar) error {
+	var secrets *corev1.Secret
+	var err error
+
+	secrets, err = GetLatestVersionedSecret(client, cr, cr.GetNamespace(), GetSplunkMonitoringConsoleDeploymentName(SplunkMonitoringConsole, cr.GetNamespace()))
+	if err != nil {
+		return err
+	}
+
+	secretName := ""
+	if secrets != nil {
+		secretName = secrets.GetName()
+	}
+
+	// create or update a regular monitoring console service
+	err = splctrl.ApplyService(client, getSplunkService(cr, &spec, SplunkMonitoringConsole, false))
+	if err != nil {
+		return err
+	}
+
+	// create or update a headless monitoring console service
+	err = splctrl.ApplyService(client, getSplunkService(cr, &spec, SplunkMonitoringConsole, true))
+	if err != nil {
+		return err
+	}
+
+	addNewURLs := true
+
+	if cr.GetObjectMeta().GetDeletionTimestamp() != nil {
+		addNewURLs = false
+	}
+
+	//_, err = ApplyMonitoringConsoleEnvConfigMap(client, cr.GetNamespace(), extraEnv, addNewURLs)
+	_, err = ApplyMonitoringConsoleEnvConfigMap(client, cr.GetNamespace(), cr.GetName(), extraEnv, addNewURLs)
+	if err != nil {
+		return err
+	}
+
+	//configMapHash to trigger the configMap change in monitoring console deployment
+	configMapHash, err := getConfigMapHash(client, cr.GetNamespace())
+	if err != nil {
+		return err
+	}
+
+	deploymentMC, err := getMonitoringConsoleDeployment(cr, &spec, SplunkMonitoringConsole, 1, configMapHash, secretName)
+	if err != nil {
+		return err
+	}
+
+	_, err = splctrl.ApplyDeployment(client, deploymentMC)
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+// GetMonitoringConsoleDeployment returns a Kubernetes Deployment object for Splunk Enterprise monitoring console instance.
+func getMonitoringConsoleDeployment(cr splcommon.MetaObject, spec *enterprisev1.CommonSplunkSpec, instanceType InstanceType, replicas int32, configMapHash string, secretName string) (*appsv1.Deployment, error) {
+	ports := splcommon.SortContainerPorts(getSplunkContainerPorts(SplunkMonitoringConsole))
+	annotations := splcommon.GetIstioAnnotations(ports)
+	var partOfIdentifier string
+	selectLabels := getSplunkLabels(cr.GetNamespace(), instanceType, partOfIdentifier) //using Namespace here so that with every CR the name should remain same Ex- splunk-<namespace>-monitoring-console
```

Comment:
Nit: Move the comment to the line above the code? And also add comments for ports, annotations(if explanation needed)?

Reply:
Moved comment and no explanation needed for ports, etc